pom.xml

```diff
@@ -181,6 +186,55 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<artifactId>maven-antrun-plugin</artifactId>
+				<version>1.7</version>
+				<executions>
+					<execution>
+						<id>jarbundler-gui</id>
+						<phase>package</phase>
+						<goals>
+							<goal>run</goal>
+						</goals>
+						<configuration>
+							<target>
+								<taskdef name="jarbundler" classname="com.ultramixer.jarbundler.JarBundler" />
```

Comment:
As for embedding the JVM, that is possible (even with the jarbundler plugin) - however, I have avoided doing that so far, because at that point, it would require being built on an OS X system.  As it currently is, you should be able to build the OS X packages even from a Windows or Linux system (an OS X app is just a certain directory structure layout).

The main reason that I used the jarbundler task is because of my familiarity with it (I've never used appbundle-maven-plugin).  It is also the "recommended" way to bundle an application by the universalJavaApplicationStub developer.  Is there a reason you'd prefer to use the appbundle-maven-plugin?